src/metabase/pulse/render/body.clj

```diff
@@ -536,32 +547,54 @@
-                        (let [[x-rowfn y-rowfn] rowfnpair]
-                          (map (juxt x-rowfn y-rowfn)
-                               (common/row-preprocess x-rowfn y-rowfn row-seq))))
+        row-seqs      (for [[row-seq [x-rowfn y-rowfn]] (map vector row-seqs rowfns)]
+                        (map (juxt x-rowfn y-rowfn)
+                             (common/row-preprocess x-rowfn y-rowfn row-seq)))
         col-seqs      (map :cols multi-data)
         first-rowfns  (first rowfns)
         [x-col y-col] ((juxt (first first-rowfns) (second first-rowfns)) (first col-seqs))
         labels        (x-and-y-axis-label-info x-col y-col viz-settings)
         names         (map :name cards)
         colors        (take (count multi-data) colors)
-        types         (map :display cards)
+        types         (replace {:scalar :bar} (map :display cards))
         settings      (->ts-viz x-col y-col labels viz-settings)
         y-pos         (take (count names) (default-y-pos viz-settings))
-        series        (join-series names colors types row-seqs y-pos)
-        image-bundle  (image-bundle/make-image-bundle
-                        render-type
-                        (js-svg/combo-chart series settings))]
-   {:attachments
-    (when image-bundle
-      (image-bundle/image-bundle->attachment image-bundle))
+        series        (join-series names colors types row-seqs y-pos)]
+     (attach-image-bundle (image-bundle/make-image-bundle render-type (js-svg/combo-chart series settings)))))
+
+(defn- multiple-scalar-series
+  [_type joined-rows _x-cols _y-cols _viz-settings]
+  ;; TODO: Extra vars could be used for color settings
+  (for [[idx row-val] (map-indexed vector joined-rows)]
+    {:name  (first row-val)
+     :color (nth colors idx)
+     :type  :bar
+     :data [row-val]
+     :yAxisPosition "left"}))
+
+(defn render-multiple-scalars
+  "When multiple scalar cards are combined, they render as a bar chart"
+  [render-type card dashcard {:keys [viz-settings] :as data}]
+  (let [viz-settings (merge viz-settings (:visualization_settings dashcard))
+        multi-res    (pu/execute-multi-card card dashcard)
+        cards        (cons card (map :card multi-res))
+        multi-data   (cons data (map #(get-in % [:result :data]) multi-res))
+        x-rows       (map :name cards) ;; Bar labels
+        y-rows       (mapcat :rows multi-data)
+        x-cols       [{:base_type :type/Text
+                       :effective_type :type/Text}]
+        y-cols       (select-keys (first (:cols data)) [:base_type :effective_type])
+        series       (multiple-scalar-series :bar (mapv vector x-rows (flatten y-rows)) x-cols y-cols viz-settings)
+        labels       (combo-label-info x-cols y-cols viz-settings)
+        settings     (->ts-viz (first x-cols) (first y-cols) labels viz-settings)]
+    (attach-image-bundle (image-bundle/make-image-bundle render-type (js-svg/combo-chart series settings)))))
```

Comment:
Could DRY up the first bits of `render-multiple-lab-chart` and `render-multiple-scalars`, but tbh I feel like the duplicated code is easier to read. There's so many individual variables floating around that there's not a great way to shorten things by much.